hack/lib/start.sh

```diff
@@ -0,0 +1,501 @@
+
+	echo "[INFO] OpenShift server health checks done at: "
+	date
+}
+readonly -f os::start::master
+
+# os::start::internal::start_all_in_one starts the OpenShift server all in one.
+# It is advised to use this function after a successful run of 'configure_os_server'
+#
+# Globals:
+#  - USE_SUDO
+#  - LOG_DIR
+#  - ARTIFACT_DIR
+#  - VOLUME_DIR
+#  - SERVER_CONFIG_DIR
+#  - USE_IMAGES
+#  - MASTER_ADDR
+#  - MASTER_CONFIG_DIR
+#  - NODE_CONFIG_DIR
+#  - API_SCHEME
+#  - API_HOST
+#  - API_PORT
+#  - KUBELET_SCHEME
+#  - KUBELET_HOST
+#  - KUBELET_PORT
+# Returns:
+#  - export OS_PID
+function os::start::internal::start_all_in_one () {
+	local sudo="${USE_SUDO:+sudo}"
+	local use_latest_images
+	if [[ -n "${USE_LATEST_IMAGES:-}" ]]; then
+		use_latest_images="true"
+	else
+		use_latest_images="false"
+	fi
+
+	echo "[INFO] Starting OpenShift server"
+	${sudo} env "PATH=${PATH}" OPENSHIFT_PROFILE=web OPENSHIFT_ON_PANIC=crash openshift start \
+		--master-config=${MASTER_CONFIG_DIR}/master-config.yaml \
+		--node-config=${NODE_CONFIG_DIR}/node-config.yaml \
+		--loglevel=4 --logspec='*importer=5' \
+		--latest-images="${use_latest_images}" \
+	&>"${LOG_DIR}/openshift.log" &
+	export OS_PID=$!
+
+	echo "[INFO] OpenShift server start at: "
+	date
+
+	wait_for_url "${API_SCHEME}://${API_HOST}:${API_PORT}/healthz" "apiserver: " 0.25 80
+	wait_for_url "${KUBELET_SCHEME}://${KUBELET_HOST}:${KUBELET_PORT}/healthz" "[INFO] kubelet: " 0.5 120
+	wait_for_url "${API_SCHEME}://${API_HOST}:${API_PORT}/healthz/ready" "apiserver(ready): " 0.25 80
+	wait_for_url "${API_SCHEME}://${API_HOST}:${API_PORT}/api/v1/nodes/${KUBELET_HOST}" "apiserver(nodes): " 0.25 80
+
+	echo "[INFO] OpenShift server health checks done at: "
+	date
+}
+readonly -f os::start::internal::start_all_in_one
+
+# os::start::internal::start_etcd starts etcd for OpenShift
+# Globals:
+#  - USE_SUDO
+#  - LOG_DIR
+#  - MASTER_CONFIG_DIR
+#  - API_SCHEME
+#  - API_HOST
+#  - ETCD_PORT
+# Arguments:
+#  None
+# Returns:
+#  - export ETCD_PID
+function os::start::internal::start_etcd() {
+	local sudo="${USE_SUDO:+sudo}"
+
+	echo "[INFO] Starting etcd"
+	${sudo} env "PATH=${PATH}" OPENSHIFT_ON_PANIC=crash openshift start etcd \
+		--config=${MASTER_CONFIG_DIR}/master-config.yaml &>"${LOG_DIR}/etcd.log" &
+	export ETCD_PID=$!
+
+	echo "[INFO] etcd server start at: "
+	date 
+
+	wait_for_url "${API_SCHEME}://${API_HOST}:${ETCD_PORT}/version" "etcd: " 0.25 80
+
+	echo "[INFO] etcd server health checks done at: "
+	date
+}
+readonly -f os::start::internal::start_etcd
+
+# os::start::internal::start_api_server starts the OpenShift API server
+# Globals:
+#  - USE_SUDO
+#  - LOG_DIR
+#  - ARTIFACT_DIR
+#  - MASTER_CONFIG_DIR
+#  - API_SCHEME
+#  - API_HOST
+#  - API_PORT
+#  - KUBELET_SCHEME
+#  - KUBELET_HOST
+#  - KUBELET_PORT
+# Arguments:
+# 1 - api server version
+# Returns:
+#  - export OS_PID
+function os::start::internal::start_api_server() {
+	local sudo="${USE_SUDO:+sudo}"
```

Comment:
I think we should bundle this into the `openshift_executable` function, we're repeating ourselves a lot with it